cloudify_aws/ecs/service.py

```diff
@@ -0,0 +1,320 @@
+        super(Service, self).__init__(
+            constants.ECS_SERVICE['AWS_RESOURCE_TYPE'],
+            constants.ECS_SERVICE['REQUIRED_PROPERTIES'],
+            client=client,
+            resource_id_key=constants.ECS_SERVICE['RESOURCE_ID_KEY'],
+        )
+
+        self.not_found_error = constants.ECS_SERVICE['NOT_FOUND_ERROR']
+
+    def get_relationship_target(self, relationships, intended_relationship):
+        target = None
+
+        for relationship in relationships:
+            if relationship.type == intended_relationship:
+                target = relationship.target
+                break
+
+        return target
+
+    def get_target_attribute(self, relationships, from_relationship,
+                             desired_attribute):
+        target = self.get_relationship_target(relationships, from_relationship)
+
+        attr = None
+
+        if target is not None:
+            attr = target.instance.runtime_properties.get(
+                desired_attribute,
+                None
+            )
+
+        return attr
+
+    def get_target_property(self, relationships, from_relationship,
+                            desired_property):
+        target = self.get_relationship_target(relationships, from_relationship)
+
+        prop = None
+
+        if target is not None:
+            prop = target.node.properties.get(
+                desired_property,
+                None
+            )
+
+        return prop
+
+    def get_arn(self, relationships, from_relationship):
+        return self.get_target_attribute(
+            relationships=relationships,
+            from_relationship=from_relationship,
+            desired_attribute=constants.EXTERNAL_RESOURCE_ID,
+        )
+
+    def get_related_elb_name(self, relationships):
+        return self.get_target_property(
+            relationships=relationships,
+            from_relationship=(
+                'cloudify.aws.relationships.ecs_service_behind_load_balancer'
+            ),
+            desired_property='elb_name',
+        )
+
+    def get_container_instance_count(self, relationships):
+        return self.get_target_attribute(
+            relationships=relationships,
+            from_relationship=(
+                'cloudify.aws.relationships.ecs_service_running_on_cluster'
+            ),
+            desired_attribute='instances',
+        )
+
+    def get_task_container_names(self, relationships):
+        return self.get_target_attribute(
+            relationships=relationships,
+            from_relationship=(
+                'cloudify.aws.relationships.ecs_service_runs_task'
+            ),
+            desired_attribute='container_names',
+        )
+
+    def create(self, args=None, **_):
+        # Don't try to do anything until the cluster has instances...
+        container_instances = self.get_container_instance_count(
+            ctx.instance.relationships,
+        )
+        if container_instances is None or len(container_instances) == 0:
```

Comment:
This is changed in the code, but apparently the new way of marking change requests doesn't update when the code is updated so I'm marking it with a comment so that I can see what I have left to deal with.